File: Api/src/Api.Domain/Entities/UserChatEntity.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Domain.Entities {
    public class UserChatEntity : BaseEntity
    {
        [Required]
        public int UserId { get; set; }

        [JsonIgnore]
        public UserEntity User { get; set; }

        [Required]
        public int ChatId { get; set; }

        [JsonIgnore]
        public ChatEntity Chat { get; set; }
    }
}

File: App/src/app/shared/entities/userChat.entity.ts
import { ChatEntity } from '../../chat/entities/chat.entity';
import { UserEntity } from '../../user/entities/user.entity';
import { BaseEntity } from './base.entity';

export interface UserChatEntity extends BaseEntity {
  userId: number;
  user?: UserEntity;
  chatId: number;
  chat?: ChatEntity;
}
